data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessor.java

```diff
@@ -163,24 +169,33 @@ public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
 
             try {
                 if (currentBufferPerBatch.getEventCount() == 0) {
-                    requestCodec.start(currentBufferPerBatch.getOutputStream(), event, codecContext);
+                    requestCodec.start(currentBufferPerBatch.getOutputStream(), event, new OutputCodecContext());
                 }
                 requestCodec.writeEvent(event, currentBufferPerBatch.getOutputStream());
                 currentBufferPerBatch.addRecord(record);
 
-                flushToLambdaIfNeeded(resultRecords, false);
+                boolean wasFlushed =  flushToLambdaIfNeeded(resultRecords, currentBufferPerBatch,
+                        requestCodec, responseCodec,futureList,false);
+
+                // After flushing, create a new buffer for the next batch
+                if (wasFlushed) {
+                    currentBufferPerBatch = lambdaCommonHandler.createBuffer(bufferFactory);
+                    requestCodec = new JsonOutputCodec(jsonOutputCodecConfig);
+                }
             } catch (Exception e) {
                 LOG.error(NOISY, "Exception while processing event {}", event, e);
                 handleFailure(e, currentBufferPerBatch, resultRecords);
-                currentBufferPerBatch.reset();
+                currentBufferPerBatch = lambdaCommonHandler.createBuffer(bufferFactory);
+                requestCodec = new JsonOutputCodec(jsonOutputCodecConfig);
```

Comment:
Why are we doing this in a catch?

Reply:
this resets the buffer and codec for the "remaining events" ; the next call to flushToLambdaIfNeeded needs it.

Reply:
Should this be in a `finally` instead?